packages/cli/src/cmds/validator/keys.ts

```diff
@@ -59,31 +57,43 @@ export async function getLocalSecretKeys(
       lockFilepath(keystorePath);
     }
 
-    const secretKeys = await Promise.all(
-      keystorePaths.map(async (keystorePath) => {
-        const keystoreStr = fs.readFileSync(keystorePath, "utf8");
+    const secretKeys: SecretKey[] = [];
+    const keyStores: Keystore[] = [];
 
-        let keystore;
-        try {
-          keystore = Keystore.parse(keystoreStr);
-        } catch (e) {
-          (e as Error).message = `Error parsing keystore at ${keystorePath}: ${(e as Error).message}`;
-          throw e;
-        }
+    for (const keystorePath of keystorePaths) {
+      try {
+        const keystoreStr = fs.readFileSync(keystorePath, "utf8");
+        keyStores.push(Keystore.parse(keystoreStr));
+      } catch (e) {
+        console.log(`Error parsing keystore file: ${path.basename(keystorePath)}: ${(e as Error).message}`);
+      }
+    }
 
-        return bls.SecretKey.fromBytes(await keystore.decrypt(passphrase));
-      })
-    );
+    for (const keystore of keyStores) {
+      // Performing decryption of keystore sequentially as concurrently calls to
+      // keystore.decrypt leads to OOM. See https://github.com/ChainSafe/lodestar/issues/4166
+      // Once the underlining issue causing the OOM is identified and fixed, the
+      // parsing can be updated to be done concurrently
+      try {
+        secretKeys.push(bls.SecretKey.fromBytes(await keystore.decrypt(passphrase)));
```

Comment:
I think we still want failure to decrypt a keystore to be a fatal error

Reply:
hmmm. That's brings us back to the comment [here](https://github.com/ChainSafe/lodestar/pull/4117#discussion_r900048292) 

So if a `password.txt` for example is placed in the keystore directory, where it's content is not that of a keystore, lodestar will die on start-up. 

I am of the opinion that as long as there is at least one keystore that was parsed successfully, lodestar should run and only log the files that failed.

The only drawback to this, is the possibility for a user to have an ill-formed keystore, start the application, fail to see the error log and wrongly assume the keystore is active on the network.

If we are to have failure to decrypt a keystore to be a fatal error, then we have to bring back a growing list of file pattern blacklist (previously deposit data file, now password.txt) so as to limit the files to be parsed that can be potentially fail.

The drawback to this is a growing list of files we expect to be wrong that can be in the keystore path, that we should always skip.

Reply:
I'm not exactly thrilled with the scope creep of this PR. I think the keystores directory and rules around that directory is something that lodestar should control. I see that directory as more of an internal directory, like the database directory, than a directory where the user can do whatever they want. Yes we allow the user to override the database dir, but we don't encourage it, and we don't provide lots of additional handling to make sure the db will be interpreted correctly.

My opinion is that lodestar should be as hard to misconfigure as possible.  The more we get into trying to figure out the intent of the user, trying to be flexible, etc, the more room there is for getting it wrong. I think sometimes that means tough love and setting firm and coherent boundaries. (In the case of our core assumptions being violated, that means a fatal error)

The original intent of this PR was that at least we can filter out things that are clearly not keystores from items found in keystoresDir.  Going further than that though, and allowing undecryptable keystores is a further step into the murky waters of subjectivity of intent.

In any case, we should definitely strengthen our docs to more clearly document the assumptions around the keystoresDir and how we intend the validator to run.

Reply:
The PR came about because of this issue https://github.com/ChainSafe/lodestar/issues/4116 where someone had `password.txt` in the keystore and that caused lodestar not to start.

I think if we want to take the approach of tough love with clear boundaries, we should only pick keystore files with known formats that can be decrypted successfully.

This was the initial implementation before you suggested that maybe having that is too strict in your comment [here](https://github.com/ChainSafe/lodestar/pull/4117#discussion_r891633914)

Based on your comment to not make file name filtering too strict, but still kill the application on start-up if the keystore cannot be decrypted properly, then we end up back at the issue: https://github.com/ChainSafe/lodestar/issues/4116

I am now more leaning to that previous implementation that picks keystore files based on known format as it sets clear boundaries on what the file format is, and what the content should be.